extendr-api/src/wrapper/externalptr.rs

```diff
@@ -82,20 +104,19 @@ impl<T: Any + Debug> ExternalPtr<T> {
     /// An ExternalPtr behaves like a Box except that the information is
     /// tracked by a R object.
     pub fn new(val: T) -> Self {
+        use std::ffi::c_void;
+        // This allocates some memory for our object and moves the object into it.
+        let boxed = Box::new(ExternalData::new(val));
         unsafe {
-            // This allocates some memory for our object and moves the object into it.
-            let boxed = Box::new(val);
-
             // This constructs an external pointer to our boxed data.
             // into_raw() converts the box to a malloced pointer.
-            let robj = Robj::make_external_ptr(Box::into_raw(boxed), r!(()));
-
+            // let robj = Robj::make_external_ptr(Box::into_raw(boxed), r!(()));
+            let robj = Robj::from_sexp(single_threaded(|| {
+                R_MakeExternalPtr(Box::into_raw(boxed) as *mut c_void, R_NilValue, R_NilValue)
```

Comment:
`Robj::make_external_ptr()` sets class name, basically second argument in this call, to type name. If you do not do that, I believe you will lose the type annotation visible in R (since from R all external ptrs will be of identical 'type').

Reply:
The type name in Rust is not considered a "stable" API. It is an arbitrary representation of a type, only intended for debug purposes, and not considered to be consistent across Rust versions or even compilation configurations.

Reply:
@iliak the idea behind this PR is to stop using tag for type checking.

Reply:
For type checking yes, but it is used not only for that.

Reply:
What else is it used for?

Reply:
Ok I take it back, it seems `tag` is not directly exposed to user in R session.

Reply:
Apart from something like this 
![image](https://github.com/extendr/extendr/assets/8782986/98c38b54-d7e1-4de8-a539-25670531fae3)

Reply:
I think that's valuable enough. But this has to happen once, and it is protected by
the `ExternalPtr`. I'll do it in a follow-up PR.

Reply:
Done.